src/dlgTriggerEditor.cpp

```diff
@@ -4403,6 +4430,115 @@ void dlgTriggerEditor::selectKeyByID(int id)
     }
 }
 
+TTrigger* dlgTriggerEditor::getTriggerFromTreeItem(QTreeWidgetItem* item)
+{
+    if (!item || !item->parent()) {
+        return nullptr;
+    }
+
+    const int triggerID = item->data(0, Qt::UserRole).toInt();
+    return mpHost->getTriggerUnit()->getTrigger(triggerID);
+}
+
+TAlias* dlgTriggerEditor::getAliasFromTreeItem(QTreeWidgetItem* item)
+{
+    if (!item || !item->parent()) {
+        return nullptr;
+    }
+
+    const int aliasID = item->data(0, Qt::UserRole).toInt();
+    return mpHost->getAliasUnit()->getAlias(aliasID);
+}
+
+TScript* dlgTriggerEditor::getScriptFromTreeItem(QTreeWidgetItem* item)
+{
+    if (!item || !item->parent()) {
+        return nullptr;
+    }
+
+    const int scriptID = item->data(0, Qt::UserRole).toInt();
+    return mpHost->getScriptUnit()->getScript(scriptID);
+}
+
+TTimer* dlgTriggerEditor::getTimerFromTreeItem(QTreeWidgetItem* item)
+{
+    if (!item || !item->parent()) {
+        return nullptr;
+    }
+
+    const int timerID = item->data(0, Qt::UserRole).toInt();
+    return mpHost->getTimerUnit()->getTimer(timerID);
+}
+
+TKey* dlgTriggerEditor::getKeyFromTreeItem(QTreeWidgetItem* item)
+{
+    if (!item || !item->parent()) {
+        return nullptr;
+    }
+
+    const int keyID = item->data(0, Qt::UserRole).toInt();
+    return mpHost->getKeyUnit()->getKey(keyID);
+}
+
+TAction* dlgTriggerEditor::getActionFromTreeItem(QTreeWidgetItem* item)
+{
+    if (!item || !item->parent()) {
+        return nullptr;
+    }
+
+    const int actionID = item->data(0, Qt::UserRole).toInt();
+    return mpHost->getActionUnit()->getAction(actionID);
+}
+
+void dlgTriggerEditor::slot_itemEdited()
+{
+    QString packageName;
+    switch (mCurrentView) {
+    case EditorViewType::cmTriggerView: {
+        if (auto trigger = getTriggerFromTreeItem(mpCurrentTriggerItem)) {
+            packageName = trigger->packageName(trigger);
+        }
+        break;
+    }
+    case EditorViewType::cmAliasView: {
+        if (auto alias = getAliasFromTreeItem(mpCurrentAliasItem)) {
+            packageName = alias->packageName(alias);
+        }
+        break;
+    }
+    case EditorViewType::cmTimerView: {
+        if (auto timer = getTimerFromTreeItem(mpCurrentTimerItem)) {
+            packageName = timer->packageName(timer);
+        }
+        break;
+    }
+    case EditorViewType::cmScriptView: {
+        if (auto script = getScriptFromTreeItem(mpCurrentScriptItem)) {
+            packageName = script->packageName(script);
+        }
+        break;
+    }
+    case EditorViewType::cmActionView: {
+        if (auto action = getActionFromTreeItem(mpCurrentActionItem)) {
+            packageName = action->packageName(action);
+        }
+        break;
+    }
+    case EditorViewType::cmKeysView: {
+        if (auto key = getKeyFromTreeItem(mpCurrentKeyItem)) {
+            packageName = key->packageName(key);
+        }
+        break;
+    }
+    case EditorViewType::cmUnknownView:
+        break;
```

Comment:
Like CodeFactor (or is that Clang) says - variables were not being handled:
```suggestion
    case EditorViewType::cmUnknownView:
        [[fallthrough]]
    case EditorViewType::cmVarsView:
        break;
```